### PR TITLE
Fix two types of issues raised by mypy - implicit optional issues and arg-type mismatch

### PR DESCRIPTION
**Pull request by gokulavasan**
Title. Here is the mypy run failing with 24 errors - https://github.com/pytorch/data/actions/runs/8428471909/job/23081016202.

With these two changes, the mypy errors are now down to 8 - https://github.com/pytorch/data/actions/runs/8428898766/job/23082258705?pr=1234

### Changes

- Add optional where it was implicit earlier
- Change ignore type to arg-type

### CODE REVIEW

**Review by ejguan — COMMENTED** [review]
Do you plan to fix all mypy error? If so, you can run `mypy --config=mypy.ini` to check them locally.

**Comment by gokulavasan** [comment]
@ejguan Yes I am trying to fix all the errors. Got it down from 24 to 8. When I run the local command that you provided above, I am getting 11 errors. I will try and get to the remaining 8 in a separate PR as they warrant a bit more debugging

**Comment by andrewkho** [comment]
LGTM but noob q: where can we see that the errors decreased?

File: torchdata/datapipes/iter/util/rows2columnar.py [hunk 1]
```diff
@@ -50,7 +50,9 @@ class Rows2ColumnarIterDataPipe(IterDataPipe[Dict]):
     """
     column_names: List[str]
 
-    def __init__(self, source_datapipe: IterDataPipe[List[Union[Dict, List]]], column_names: List[str] = None) -> None:
+    def __init__(
+        self, source_datapipe: IterDataPipe[List[Union[Dict, List]]], column_names: Optional[List[str]] = None
+    ) -> None:
```
Comment (andrewkho) [inline torchdata/datapipes/iter/util/rows2columnar.py:55]: Another n00b q: what formatter are you using in torchdata vs torch.utils.data? Is linting part of CI on the github side?

Comment (gokulavasan) [reply torchdata/datapipes/iter/util/rows2columnar.py:55]: Formatter being used - black - https://github.com/pytorch/data/blob/4725a01ce38c646b6273acacb9df895b3d9c3f66/.pre-commit-config.yaml#L28-L34.

linting part of CI is on the github side. I think there is an explicit exclusion of it in phabricator.

**Review by ejguan — APPROVED** [review]

**Pull request closed without merging**